asyncpg/serverversion.py

```diff
@@ -16,14 +16,28 @@ def split_server_version_string(version_string):
     parts = version_string.strip().split('.')
     if not parts[-1].isdigit():
         # release level specified
-        level = parts[-1].rstrip('0123456789').lower()
-        serial = parts[-1][level:]
-        versions = [int(p) for p in parts[:-1]][:3]
+        lastitem = parts[-1]
+        levelpart = lastitem.rstrip('0123456789').lower()
+        if levelpart != lastitem:
+            serial = int(lastitem[len(levelpart):])
+        else:
+            serial = 0
+
+        level = levelpart.lstrip('0123456789')
+        if level != levelpart:
+            parts[-1] = levelpart[:-len(level)]
+        else:
+            parts[-1] = 0
     else:
         level = 'final'
         serial = 0
-        versions = [int(p) for p in parts][:3]
 
+    if int(parts[0]) >= 10:
+        # Since Postgres v10 there are only "major" and "micro" parts
```

Comment:
I agree that from point of view of just tuples it seems better, but...
It is a preparation for "ServerVersion" named tuple, and logically the last digit in "10.1" means "micro", and "minor" is considered to be 0.
In fact "10.1" is an analogue of current "9.5.1", changes between "micro" versions are not so incompatible, so usually you don't need to write "if server_version < (10,1):".
Therefore semantic looks as "if server_version < (10,0):" instead of "if server_version < (10,):".

If you still insist on removing this block, I do it after your next request.